WcaOnRails/app/webpacker/components/RolesTab/index.jsx

```diff
@@ -6,77 +6,133 @@ import {
   List,
 } from 'semantic-ui-react';
 import useLoadedData from '../../lib/hooks/useLoadedData';
+import useSaveAction from '../../lib/hooks/useSaveAction';
 import {
   roleListUrl,
+  currentUserUrl,
+  teamUrl,
 } from '../../lib/requests/routes.js.erb';
 import Errored from '../Requests/Errored';
 import Loading from '../Requests/Loading';
 import RoleForm from './RoleForm';
+import I18n from '../../lib/i18n';
 
-function RoleFormModal({
-  trigger, userId, isActiveRole, sync,
-}) {
-  const [open, setOpen] = React.useState(false);
-
-  return (
-    <Modal
-      size="fullscreen"
-      onClose={() => {
-        setOpen(false);
-        sync();
-      }}
-      onOpen={() => setOpen(true)}
-      open={open}
-      trigger={trigger}
-    >
-      <Modal.Content>
-        <RoleForm
-          userId={userId}
-          isActiveRole={isActiveRole}
-        />
-      </Modal.Content>
-    </Modal>
-  );
-}
+// let i18n-tasks know the key is used
+// i18n-tasks-use t('enums.user.role_status.trainee_delegate')
+// i18n-tasks-use t('enums.user.role_status.candidate_delegate')
+// i18n-tasks-use t('enums.user.role_status.delegate')
+// i18n-tasks-use t('enums.user.role_status.senior_delegate')
+// i18n-tasks-use t('enums.user.role_status.member')
+// i18n-tasks-use t('enums.user.role_status.senior_member')
+// i18n-tasks-use t('enums.user.role_status.leader')
 
 export default function RolesTab({ userId }) {
+  const currentUser = useLoadedData(currentUserUrl);
   const {
     data, sync, loading, error,
   } = useLoadedData(roleListUrl(userId));
+  const { save, saving } = useSaveAction();
+
+  const [open, setOpen] = React.useState(false);
+  const [currentUserRoles, setCurrentUserRoles] = React.useState(null);
+  const [currentUserFetchError, setCurrentUserFetchError] = React.useState(null);
+
+  const activeNonHiddenRoles = React.useMemo(
+    () => {
+      if (data) {
+        return data.activeRoles.filter((role) => role.group.is_hidden === false);
```

Comment:
I'm not familiar with hidden roles, though they sound familiar. How "hidden" are they meant to be? If they're actually secret and shouldn't be leaked, then the data endpoint should not be returning them at all right?

Reply:
Currently the data endpoint returns only delegate roles, and we are not returning any hidden roles.
I did this filter just for a double safety. In future, the endpoint will return hidden roles as well like probation, etc. I'll make sure not to send hidden roles from there once I implement that.